src/main/scala/tech/cryptonomic/conseil/tezos/ApiOperations.scala

```diff
@@ -1,29 +1,40 @@
 package tech.cryptonomic.conseil.tezos
 
 import slick.jdbc.PostgresProfile.api._
-import tech.cryptonomic.conseil.generic.chain.{DataOperations, DataTypes, MetadataOperations}
-import tech.cryptonomic.conseil.generic.chain.DataTypes.{
-  Field,
-  FormattedField,
-  OperationType,
-  OrderDirection,
-  Predicate,
-  Query,
-  QueryOrdering,
-  QueryResponse,
-  SimpleField
-}
+import tech.cryptonomic.conseil.generic.chain.{DataOperations, MetadataOperations}
+import tech.cryptonomic.conseil.generic.chain.DataTypes.{Field, FormattedField, Predicate, Query, QueryResponse, SimpleField}
+import tech.cryptonomic.conseil.tezos.ApiOperations.{AccountResult, BlockResult, OperationGroupResult}
 import tech.cryptonomic.conseil.tezos.TezosTypes.{AccountId, BlockHash}
 import tech.cryptonomic.conseil.tezos.{TezosDatabaseOperations => TezosDb}
 import tech.cryptonomic.conseil.util.DatabaseUtil
 
 import scala.concurrent.{ExecutionContext, Future}
 
+
+object ApiOperations {
+  case class BlockResult(block: Tables.BlocksRow, operation_groups: Seq[Tables.OperationGroupsRow])
+  case class OperationGroupResult(operation_group: Tables.OperationGroupsRow, operations: Seq[Tables.OperationsRow])
+  case class AccountResult(account: Tables.AccountsRow)
+
+
+  /** Sanitizes string to be viable to paste into plain SQL */
+  def sanitizeForSql(str: String): String = {
+    val supportedCharacters = Set('_', '.', '+', ':', '-', ' ')
+    str.filter(c => c.isLetterOrDigit || supportedCharacters.contains(c))
+  }
+
+  /** Sanitizes datePart aggregate function*/
+  def sanitizeDatePartAggregation(str: String): String = {
+    val supportedCharacters = Set('Y', 'M', 'D', 'A', '-')
+    str.filterNot(c => c.isWhitespace || !supportedCharacters.contains(c))
+  }
+}
+
 /**
   * Functionality for fetching data from the Conseil database.
   */
-object ApiOperations extends DataOperations with MetadataOperations {
-
+class ApiOperations extends DataOperations with MetadataOperations {
+  import ApiOperations._
   lazy val dbReadHandle: Database = DatabaseUtil.conseilDb
```

Comment:
What about passing this in as a parameter?

Reply:
passing DB as a parameter would require adding DB in the tests which don't use database(only stubs/mocks of the ApiOperations), or maybe do you have other proposal how to handle it?

Reply:
I guess the parameterization would look something like `DBIOAction[T] => Future[T]` with no reference to a `Database`.

I know by direct handling this that using the latter as a generic definition is terrible to mock. Instead you can easily get away with it, using the "function" parameter.

Still, do you think it might improve the situation? I can't be sure from this point.

Reply:
I'm not sure what you have in mind. Parametrising by `[T]` would require creating parameter for class and `[T]` would be different for each method. Or am I missing something and I misunderstood your idea?